src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.Arrays;
```

Comment:
The use of `Arrays` utility class is against the checklist guidelines. Consider implementing array resizing without using `Arrays.copyOf`.